pkg/client/cache/delta_fifo.go

```diff
@@ -334,26 +336,42 @@ func (f *DeltaFIFO) Replace(list []interface{}) error {
 				continue
 			}
 		}
-		if err := f.queueActionLocked(Deleted, DeletedFinalStateUnknown{k}); err != nil {
+		v, exists, err := f.knownObjects.GetByKey(k)
+		if err != nil || !exists {
+			v = nil
+			glog.Infof("Unable to lookup key %v returned by list, ignoring", k)
+		}
+		if err := f.queueActionLocked(Deleted, DeletedFinalStateUnknown{k, v}); err != nil {
 			return err
 		}
 	}
 	return nil
 }
 
-// A KeyLister is anything that knows how to list its keys.
-type KeyLister interface {
+// A KeyLookup is anything that knows how to list its keys.
+type KeyLookup interface {
 	ListKeys() []string
+	GetByKey(key string) (interface{}, bool, error)
 }
 
-// KeyListerFunc adapts a raw function to be a KeyLister.
-type KeyListerFunc func() []string
+// KeyLookupFunc adapts a raw function to be a KeyLookup.
+type KeyLookupFunc func() []string
```

Comment:
yeah I don't believe it is used anywhere else so I've moved it to delta_fifo_test